### ISSUE TITLE
New DataTypes

### ISSUE THREAD

**Issue by VaughnRupp** [opensourceautomation/Open-Source-Automation#361]
New data types should be considered to add more custom validation and formatting.

1 Phone Number
2 Email

Validation coded to support them in the Web UI and possibly object_property_set proc

**Comment by VaughnRupp** [comment]
Currency maybe too.   Which would save as a float, but could format neatly then.

**Comment by KHerron** [comment]
Yes, thank you. The validation code is already there. I will news to ads the float.


Sent via the Samsung GALAXY S®4, an AT&T 4G LTE smartphone

-------- Original message --------

**Comment by KHerron** [comment]
Oh, and maybe a Percent. Jest like currency, it would save as a float, but could be formatted with a percent sign.

**Comment by KHerron** [comment]
I've got another data type we need to consider: IMAGE
It appears Images are current set to FILE.

Which is OK, but we could do more specific task if we separated Image from File since the image is actually inside the database.

Also in my Exporting of Screens, It would be nice to look for IMAGE type properties other than FILE then determine if it is actually a file or an image.

**Comment by KHerron** [comment]
New Data Types have been added and committed
